### PR TITLE
 Changes to make scopes type as Set

### PR DESCRIPTION
**Pull request by kreedula**
- Also added `addDefaultScopes` util method to base controller
- Setting accesstoken target from request scopes.
- Fixes : https://github.com/AzureAD/microsoft-authentication-library-common-for-android/issues/384

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/internal/controllers/BaseController.java [hunk 1]
```diff
@@ -368,6 +356,16 @@ protected boolean accessTokenIsNull(ICacheRecord cacheRecord) {
         return null == cacheRecord.getAccessToken();
     }
 
+    protected void addDefaultScopes(final OperationParameters operationParameters){
+        final Set<String> requestScopes = operationParameters.getScopes();
+        requestScopes.add(OPEN_ID_SCOPE);
+        requestScopes.add(OFFLINE_ACCESS_SCOPE);
+        requestScopes.add(PROFILE_SCOPE);
+        // sanatize empty and null scopes
```
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/internal/controllers/BaseController.java]: nit: `sanitize`

**Comment by kreedula** [comment]
Thanks @iambmelt , Fixed the typo

File: common/src/main/java/com/microsoft/identity/common/internal/controllers/BaseController.java [hunk 2]
```diff
@@ -77,6 +75,10 @@
 
     private static final String TAG = BaseController.class.getSimpleName();
 
+    private static final String OPEN_ID_SCOPE = "openid";
```
Comment (rapong) [inline common/src/main/java/com/microsoft/identity/common/internal/controllers/BaseController.java]: nit: I suggest putting these values in AuthenticatorConstants, as these values is also being used somewhere else (PRTController)

Comment (kreedula) [reply common/src/main/java/com/microsoft/identity/common/internal/controllers/BaseController.java]: Thanks for the suggestion @rapong . Addressed it

**Comment by kreedula** [comment]
Thanks @iambmelt

**Pull request merged**